csharp/ql/src/semmle/code/csharp/Type.qll

```diff
@@ -999,5 +999,63 @@ class TypeMention extends @type_mention {
 @type_or_ref getTypeRef(@type type) {
   result = type
   or
-  typeref_type(result, type)
+  type = result.(TypeRefs::TypeRef).getCanonicalType()
+}
+
+/**
+ * Module for reasoning about typerefs.
+ */
+private module TypeRefs {
```

Comment:
Sounds like a good idea - mainly because the C# `.qll` files tend to be too large and should in general be split up. I'll call it `TypeRef.qll` since the precedent is to use the singular form when naming `.qll` files.